test/system/src/main/java/io/pravega/test/system/framework/services/marathon/PravegaControllerService.java

```diff
@@ -122,18 +121,18 @@ private App createPravegaControllerApp() {
 
         String componentCode = "controller";
 
-        //set env
+        // These properties are defined in "io.pravega.controller.util.Config".
         String controllerSystemProperties = "-Xmx512m" +
-                setSystemProperty(Config.PROPERTY_ZK_URL.getFullName(componentCode), zk) +
-                setSystemProperty(Config.PROPERTY_RPC_HOST.getFullName(componentCode), this.id + ".marathon.mesos") +
-                setSystemProperty(Config.PROPERTY_RPC_PORT.getFullName(componentCode), String.valueOf(CONTROLLER_PORT)) +
-                setSystemProperty(Config.PROPERTY_SERVICE_PORT.getFullName(componentCode), String.valueOf(CONTROLLER_PORT)) +
-                setSystemProperty(Config.PROPERTY_REST_PORT.getFullName(componentCode), String.valueOf(REST_PORT)) +
+                setSystemProperty(String.format("%s.%s", componentCode, "zk.connect.uri"), zk) +
+                setSystemProperty(String.format("%s.%s", componentCode, "service.rpc.published.host.nameOrIp"), this.id + ".marathon.mesos") +
```

Comment:
There is a lot of code repetition here. Extracting this out to a method that takes only the config name and the value and assumes the component code can simplify this:

```
setConfig("zk.connect.uri", zk)
```

p.s.: The name of the method `setSystemProperty()` is also misleading. It doesn't really "set the system property", but rather returns a newly created string representing the system property (`-Dprop=value`). But, that's preexisting code, and you may or may not want to  change that too.